ntoskrnl/io/iomgr/deviface.c

```diff
@@ -412,6 +633,183 @@ IopOpenInterfaceKey(IN CONST GUID *InterfaceClassGuid,
     return Status;
 }
 
+/*++
+ * @name IoGetDeviceInterfaceAlias
+ * @implemented
+ *
+ * Returns the alias device interface of the specified device interface
+ * instance, if the alias exists.
+ * Documented in WDK.
+ *
+ * @param SymbolicLinkName
+ *        Pointer to a string which identifies the device interface instance
+ *
+ * @param AliasInterfaceClassGuid
+ *        See WDK
+ *
+ * @param AliasSymbolicLinkName
+ *        See WDK
+ *
+ * @return Three different NTSTATUS values in case of errors, and STATUS_SUCCESS
+ *         otherwise (see WDK for details)
+ *
+ * @remarks Must be called at IRQL = PASSIVE_LEVEL in the context of a system thread
+ *
+ *--*/
+NTSTATUS
+NTAPI
+IoGetDeviceInterfaceAlias(
+    _In_ PUNICODE_STRING SymbolicLinkName,
+    _In_ CONST GUID *AliasInterfaceClassGuid,
+    _Out_ PUNICODE_STRING AliasSymbolicLinkName)
+{
+    UNICODE_STRING AliasSymbolicLink = {0, 0, NULL};
+    UNICODE_STRING PrefixString;
+    UNICODE_STRING AliasGuidString;
+    UNICODE_STRING DeviceString, ReferenceString;
+    PKEY_VALUE_FULL_INFORMATION kvInfo;
+    HANDLE DeviceKey, AliasInstanceKey;
+    PVOID Buffer;
+    USHORT i;
+    NTSTATUS Status;
+
+    RtlInitEmptyUnicodeString(&PrefixString, NULL, 0);
```

Comment:
Why do you initialize some strings with `RtlInitEmptyUnicodeString` and some strings with a direct assignment?

Reply:
@Extravert-ir it was just @ThFabba's suggestion. It initializes `NULL` unicode string properly for usage. :slightly_smiling_face: